lib/blocks.php

```diff
@@ -93,3 +93,104 @@ function do_blocks( $content ) {
+ *
+ * @since 1.1.0
+ *
+ * @param string $content The post content.
+ *
+ * @return array Array of block data.
+ */
+function gutenberg_add_blocks_to_post_resource( $content ) {
+	$registry = WP_Block_Type_Registry::get_instance();
+	$blocks   = gutenberg_parse_blocks( $content );
+	$data     = array();
+
+	// Loop thru the blocks, adding rendered content when available.
+	foreach ( $blocks as $block ) {
+		$block_name  = isset( $block['blockName'] ) ? $block['blockName'] : null;
+		$attributes  = is_array( $block['attrs'] ) ? $block['attrs'] : null;
+		$raw_content = isset( $block['rawContent'] ) ? $block['rawContent'] : null;
+
+		// Skip block if we didn’t get a valid block name.
+		if ( null === $block_name ) {
+			continue;
+		}
+
+		// Set up rendered content, if available.
+		$block['renderedContent'] = null;
+		$block_type = $registry->get_registered( $block_name );
+		if ( null !== $block_type ) {
+			$block['renderedContent'] = $block_type->render( $attributes, $raw_content );
+		}
+
+		// Add the item data.
+		$data[] = array(
+			'type'       => $block_name,
+			'attributes' => $attributes,
+			'content'    => $block['rawContent'],
+			'rendered'   => $block['renderedContent'],
+		);
+	}
+
+	return $data;
+}
+
+/**
+ * Attach a post's block data callback to the REST API response.
+ *
+ * @since 1.1.0
+ *
+ * @param string | array $post_types Post type, or array of post types.
+ */
+function gutenberg_attach_block_response_callback( $post_types ) {
+	if ( empty( $post_types ) ) {
+		$post_types = 'post';
+	}
+	if ( ! is_array( $post_types ) ) {
+		$post_types = array( $post_types );
+	}
+	foreach ( $post_types as $post_type ) {
+
+		/**
+		 * Filter whether a post type has its blocks data added the REST API response content.
+		 *
+		 * @since 1.1.0
+		 *
+		 * @param bool   $blocks_show_in_rest Whether to show blocks in the REST API response.
+		 * @param string $post_type           The post type.
+		 */
+		if ( apply_filters( 'gutenberg_add_blocks_to_rest_for_post_type', true, $post_type ) ) {
+			add_filter( 'rest_prepare_' . $post_type, 'gutenberg_extract_blocks_from_post_content', 10, 3 );
+		}
+	}
+}
+gutenberg_attach_block_response_callback( array( 'post', 'page' ) );
+
+/**
+ * Attach a post's block data to the REST API response.
+ *
+ * @since 1.1.0
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Post          $post     The Post object.
+ *
+ * @return WP_REST_Response $response The filtered response object.
+ */
+function gutenberg_extract_blocks_from_post_content( $response, $post ) {
+	if ( ! $post ) {
+		return $response;
+	}
+
+	// Extract the block data from the post content.
+	$blocks = gutenberg_add_blocks_to_post_resource( $post->post_content );
+
+	// Add block data to the response as part of 'content'.
+	$content           = $response->get_data( 'content' );
+	$content['blocks'] = $blocks;
+	$response->set_data( array( 'content' => $content ) );
```

Comment:
Maybe related to my previous content, I'm seeing that the post entity is now wrapped inside a `content` key in the REST response:

![response](https://user-images.githubusercontent.com/1779930/30892180-20ef7cbe-a305-11e7-834d-712277cce06a.png)

Reply:
Thanks for the feedback & testing @aduth, I'll review these issues.

Reply:
@aduth Oops! fixed that in  https://github.com/WordPress/gutenberg/pull/2649/commits/d395197eaaf381d285c7afdc540e384712b71ba0